lib/picam/camera.ex

```diff
@@ -33,11 +40,19 @@ defmodule Picam.Camera do
 
   def handle_info({_, {:data, jpg}}, state) do
     Task.start(fn -> dispatch(state.requests, jpg) end)
-    {:noreply, %{state | requests: []}}
+    {:noreply, %{state | requests: [], offline: false}}
+  end
+
+  def handle_info(:reconnect_port, state) do
+    executable = Path.join(:code.priv_dir(:picam), "raspijpgs")
+    port = Port.open({:spawn_executable, executable}, [{:packet, 4}, :use_stdio, :binary, :exit_status])
+
+    {:noreply, %{state | port: port}}
   end
 
   def handle_info({_, {:exit_status, _}}, state) do
-    {:stop, :unexpected_exit, state}
+    Process.send_after(self(), :reconnect_port, 10_000)
```

Comment:
Let's provide a parameter for the restart interval so it's a little easier to change if needed.  My next comment will touch on how I would do this.